lib/solidus_bactracs/api/shipment_serializer.rb

```diff
@@ -23,7 +23,7 @@ def call(sguid: nil)
               xml.sGuid                       sguid
               xml.NewRMA {
                 xml.RMANumber                 @shipment.number
-                xml.RMATypeName               @config.default_rma_type
+                xml.RMATypeName               rma_type
```

Comment:
considering that this is the core place that the change needs to happen, I'm looking at 3 patterns for doing it, the 1 is what is in this PR
1. pass something that represents the value 'down the chain' of the call stack.  Here, `true/false` is passed from the 'root' at `ScheduleShipmentSyncsJob`, and then at `bulk_create_orders` it is transformed into `W` or `4`.  That won't work once we need to support an additional type.
2. in the serializer here, use a new proc `evaluate_rma_type` that determines it; following the same pattern as `@error_handler` and `@api_shipment_matcher` etc.  That might invoke N+1 queries or more. 
3. overload `Spree::Shipment` with a decorator like `rma_type` and set it on all shipments at the root job.  The call stack passes whole Shipment objects which will retain the data.  Robust if we also fallback, in the serializer, and call the `evaluate_rma_type` from #2, in the case that `rma_type` wasn't set/passed.  This way requires some checked to ensure that the Shipment is decorated, but that's then a 'boundary' between the gem's generic andlor code, and our Suvie implementation.

Reply:
If #3 works out, then this PR is can make later be reverted. 

* the passing of `is_trade_up` boolean isn't needed
* the new job is moved to SC API
* the conditional on Bactracs Sync Verify isn't needed?

the only thing to then add here is

* register a proc in the configs, named like `evaluate_rma_type`
* the Shipment decorator/field
* validation or warning when that rma_type field not present

then in the SC API PR there will be
* changes to the existing job
* the new job, moved from here
* the `evaluate_rma_type` proc definition

Reply:
#19 did this 👍